### PR TITLE
[DependencyInjection] Fix fetching lazy non-shared services multiple times with as files `true`

### LINKED ISSUE #50975: Fetching of lazy non-shared services sometimes broken
### Symfony version(s) affected

6.3.*

### Description

This is a close-to-absurd edge case, but still it occurred in my project so I report it.

When fetching a service from the service container that is public, non-shared, and lazy, twice (!), the following exception is thrown:

```
Symfony\Component\DependencyInjection\Container::get(): Return value must be of type ?object, string returned
```

On Symfony 6.2 or earlier versions, this exception was not thrown. It also doesn't seem to occur if the service is injected in a constructor instead of being pulled from the container.

Thank you for a look!

### How to reproduce

Define a service as follows:

```yaml
App\Service\SomeService:
    public: true
    shared: false
    lazy: true
```

Then fetch it from the service container twice:

```php
$this->container->get(SomeService::class);
$this->container->get(SomeService::class);
```

### Possible Solution

_No response_

### Additional Context

The problem occurs in the container's factory method for `SomeService`. On the second call, `$lazyLoad` contains the service ID instead of the service instance.

### PR DESCRIPTION
**Pull request by HypeMC**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #50975
| License       | MIT
| Doc PR        | -

Followup to #50985, fixes case when `as_files` is `true`.

This particular problem exists only on v6.3 and was introduced in #48469.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php [hunk 1]
```diff
@@ -772,7 +754,7 @@ public function testNonSharedLazy()
         $container = new ContainerBuilder();
 
         $container
-            ->register('foo', Foo::class)
+            ->register('foo', \Bar\FooLazyClass::class)
```
Comment (HypeMC) [inline src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php:757]: A minor mistake I made in my original PR, doesn't really change anything, but it makes the test a bit more consistent.

File: src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php [hunk 2]
```diff
@@ -803,6 +785,48 @@ public function testNonSharedLazy()
...
+        $stringDump = print_r($dumps, true);
+        $this->assertStringMatchesFormatFile(
+            self::$fixturesPath.'/php/services_non_shared_lazy_as_files.txt',
+            '\\' === \DIRECTORY_SEPARATOR ? str_replace("'.\\DIRECTORY_SEPARATOR.'", '/', $stringDump) : $stringDump
+        );
+
+        $lastDump = array_pop($dumps);
+        foreach (array_reverse($dumps) as $dump) {
+            eval('?>'.$dump);
+        }
```
Comment (HypeMC) [inline src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php:817]: Is there a better way to do this?

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php:817]: Nah, that's fine

**Comment by nicolas-grekas** [comment]
Thank you @HypeMC.

**Pull request merged**